binaries/qsfpinfo/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+## Copyright(c) 2024, Silciom Denmark A/S
+##
+## Redistribution  and  use  in source  and  binary  forms,  with  or  without
+## modification, are permitted provided that the following conditions are met:
+##
+## * Redistributions of  source code  must retain the  above copyright notice,
+##   this list of conditions and the following disclaimer.
+## * Redistributions in binary form must reproduce the above copyright notice,
+##   this list of conditions and the following disclaimer in the documentation
+##   and/or other materials provided with the distribution.
+## * Neither the name  of Intel Corporation  nor the names of its contributors
+##   may be used to  endorse or promote  products derived  from this  software
+##   without specific prior written permission.
+##
+## THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+## AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+## IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+## ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+## LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+## CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+## SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+## INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+## CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
+
+opae_add_executable(TARGET qsfpinfo
+    SOURCE
+        main.c
+        cmis_wrap.c
+        sff-common_wrap.c
+    LIBS
+        argsfilter
+        opae-c
+        m
+    COMPONENT toolqsfpinfo
+)
+
+target_include_directories(qsfpinfo
+    PRIVATE
+        ${OPAE_LIB_SOURCE}/argsfilter
+)
+
+include(ExternalProject)
+
+ExternalProject_Add(
+    ethtool_project
+    GIT_REPOSITORY "https://git.kernel.org/pub/scm/network/ethtool/ethtool.git"
+    GIT_TAG "v6.7"
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+    UPDATE_COMMAND ""
+)
+
+ExternalProject_Get_Property(ethtool_project SOURCE_DIR)
```

Comment:
The code used is normally compiled into the ethertool binary. The qsfpinfo program calls a static function internally in the code. So there is no library providing that interface. Since the dfl drivers do not create a Linux network interface we cannot use ethertool directly.

ExternalProject_Add() is also used in other places in opae-sdk.

Reply:
@matthew-gerlach may no more about why the drivers do not implement "a Linux netlink interface". The qsfp-mem driver only implements QSFP handling. I know that the HSSI could not be accepted by Linux community as a network interface because you cannot send and receive packets. So HSSI is handled in user space.

Reply:
To be a valid, accepted "Linux netlink interface", the device must transmit and receive packets between the network connection and memory. The HSSI and QSFP subsystems are just pieces of the pipeline of full netlink device.

Reply:
I have made the printing of QSFP information, thus importing from Ethertool, dependent on cmake option OPAE_WITH_QSFPINFO_QSFPPRINT, which is OFF by default. qsfpinfo will thus by default only print if the QSFP modules are present or not. In this way the qsfpinfo binary is always part of the packages built.

Reply:
Hi @pcolberg, I have rebased the branch to current master. Now all checks pass. Please review again.